cmd/theatre-envconsul/main.go

```diff
@@ -133,6 +133,32 @@ func mainError(ctx context.Context, command string) (err error) {
 			}
 		}
 
+		var filePaths = environment{}
+		var pathTBD = "PATH_TBD"
+
+		// Rewrite vault-file: envs to vault: envs. Store the paths to which they should
+		// be written to in filePaths
+		for key, value := range env {
+			if strings.HasPrefix(value, "vault-file:") {
+				// determine if we define a path at which to place the file
+				trimmed := strings.TrimPrefix(value, "vault-file:")
+				split := strings.SplitN(trimmed, ":", 2)
+				switch len(split) {
+				case 2: // path and key
+					filePaths[key] = split[1]
+					env[key] = fmt.Sprintf("vault:%s", split[0])
+				case 1: // just key
+					filePaths[key] = pathTBD
+					env[key] = fmt.Sprintf("vault:%s", trimmed)
+				default:
```

Comment:
Given what's returned from `SplitN`, I don't think that we can ever actually reach this?

Reply:
Ah yea, no `:` present will evaluate to `case 1`, which is what we want. I suppose with `SplitN(trimmed, ":", -1)` we have validation for too many `':'`'s? Or do you think it should just be dropped?

Reply:
I think arguably we could just drop it. We may want some separate validation for when the input string is simply `vault-file:` with no extra components though. If that's set, then we should probably just throw an error and exit